download-test_readline.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+# From ruby/ruby, copies readline test files & tool/lib files
+# Replicates testing used in Ruby master
+
+require 'uri'
+require 'net/http'
+require 'openssl'
+require 'fileutils'
+
+module HTTPS_DL
+
+  # number of concurrent connections
+  CONNECTIONS = 4
+
+  HOST = "https://raw.githubusercontent.com"
+  URI_GH = URI HOST
+  BASE = "ruby/ruby/master"
+
+  # 1st - ruby/ruby dir  , 2nd - reline dir    , 3rd - filename
+  FILES = [
+    ["test/readline"     , "test/ext/readline" , "helper.rb"                ],
+    ["test/readline"     , "test/ext/readline" , "test_readline.rb"         ],
+    ["test/readline"     , "test/ext/readline" , "test_readline_history.rb" ],
+    ["tool/lib"          , "tool/lib"          , "colorize.rb"              ],
+    ["tool/lib"          , "tool/lib"          , "envutil.rb"               ],
+    ["tool/lib"          , "tool/lib"          , "find_executable.rb"       ],
+    ["tool/lib"          , "tool/lib"          , "leakchecker.rb"           ],
+    ["tool/lib/minitest" , "tool/lib/minitest" , "unit.rb"                  ],
+    ["tool/lib/test"     , "tool/lib/test"     , "unit.rb"                  ],
+    ["tool/lib/test/unit", "tool/lib/test/unit", "assertions.rb"            ],
+    ["tool/lib/test/unit", "tool/lib/test/unit", "core_assertions.rb"       ],
+    ["tool/lib/test/unit", "tool/lib/test/unit", "parallel.rb"              ],
+    ["tool/lib/test/unit", "tool/lib/test/unit", "testcase.rb" ]
+  ]
+
+  class << self
+
+    def run
+      files = FILES
+
+      dirs = FILES.map { |l| l[1] }.uniq
+      dirs.each { |dir| FileUtils.mkdir_p("./#{dir}") unless Dir.exist? dir }
+
+      @connections = []
+      @mutex = Mutex.new
+
+      CONNECTIONS.times do
+        @connections << Thread.new do
+          Net::HTTP.start(URI_GH.host, URI_GH.port, :use_ssl => true, :verify_mode => OpenSSL::SSL::VERIFY_PEER) do |http|
+            done = false
+            loop do
+              path = nil
+              dir = nil
+              file = nil
+              @mutex.synchronize do
+                if files.empty?
+                  done = true
+                else
+                  path, dir, file = files.shift
+                end
+              end
+              break if done
+
+              uri = URI("#{HOST}/#{BASE}/#{path}/#{file}")
+              req = Net::HTTP::Get.new uri.request_uri
+              http.request req do |res|
+                unless res.message.strip == 'OK'
+                  STDOUT.puts "Can't download #{path}/#{file} from #{HOST}/#{BASE}/"
+                  exit 1
+                end
+                File.open("./#{dir}/#{file}", 'wb') { |io| io.write res.body }
```

Comment:
You can use `File.binwrite`.

Reply:
Thanks.  Must have been rushed.  `File.open` can misbehave with windows,  I never use it.

Reply:
@nobu

Thank you for taking the time to review the code.  Your knowledge of Ruby is exceptional, and the Ruby organization is lucky to have your contributions.